sunflower/__main__.py

```diff
@@ -17,9 +17,15 @@
 #   You should have received a copy of the GNU General Public License
 #   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from __future__ import absolute_import, print_function
+
 import os
 import sys
 
+if sys.version_info[0] == 2:
+	reload(sys)
+	sys.setdefaultencoding('utf8')
+	sys.path.append(os.path.dirname(__file__)) # Not necessary for us, but it might help old plugins
```

Comment:
I'm not entirely sure if this line is a good idea in the first place.

Something like this *would* work for Python 3, but then we'd have to rename the `queue` module for one to avoid shadowing the stdlib module. In the end it's a bit of a stylistic choice whether to pollute the global package namespace with internal module names or not. Putting everything in one top-level package makes it clear what's part of sunflower and what isn't.

Reply:
To be honest I am not liking it either, and your approach with `__main__.py` seems to work rather well. I'd rather get rid of this code if it works with Py2. My general issue is relative imports. I really don't like those as it's not always clear from where we are importing things and it might be just problem waiting to happen.

I'd rather have
```python
from sunflower.gui.input_dialog import InputDialog
```
Than this
```python
from ..input_dialog import InputDialog
```

Reply:
Okay, so get rid of that little section.

And make everything absolute? Or keep `from .` but get rid of all `from ..`?

You should have write access to this branch btw if I understand how GitHub does things. If you want to edit things here yourself before you're ready to merge. (Or I'll change what we discussed later)